src/internal/connector/discovery/api/users.go

```diff
@@ -220,3 +326,31 @@ func validateUser(item any) (models.Userable, error) {
 
 	return m, nil
 }
+
+func convertInterfaceToString(ctx context.Context, data interface{}, userInfo *UserInfo) string {
+	var (
+		ok          bool
+		dataPointer *string
+		value       string
+	)
+
+	dataPointer, ok = data.(*string)
+	if !ok {
+		logger.Ctx(ctx).Infof("error getting mailboxSettings")
+
+		userInfo.ErrGetMailBoxSetting = "not found"
+
+		return ""
+	}
+
+	value, ok = ptr.ValOK(dataPointer)
+	if !ok {
```

Comment:
ok doesn't check for zero len strings.  Is this a better check?
```suggestion
	value := ptr.ValdataPointer)
	if len(value) == 0 {
```

Reply:
there is a possibility that few values are blank and length will be zero in that case. We are fine with it. Only scenario we want to avoid is that - the code should not break with a nil pointer exception  if some field is unset.